lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyXSDURIResolver.java

```diff
@@ -69,4 +101,48 @@ public Map<String, String> getExternalGrammarLocation(URI fileURI) {
     return externalGrammar;
   }
 
+  /**
+   * Generate the schema file for a LibertyWorkspace using the ws-schemagen.jar in the corresponding Liberty installation
+   * @param libertyWorkspace
+   * @param schemaGenJarPath
+   * @return Path to generated schema file.
+   */
+  private String generateServerSchemaXsd(LibertyWorkspace libertyWorkspace, Path schemaGenJarPath) {
+    //java -jar path/to/ws-schemagen.jar path/to/workspace/.libertyls/libertySchema.xsd
+    File tempDir = LibertyUtils.getTempDir(libertyWorkspace.getWorkspaceURI().toString());
+
+    //TODO: (?) Add subfolders to tempDir: schema/xsd/liberty/server.xsd
+    File xsdDestFile = new File(tempDir, "server.xsd");
+    if (libertyWorkspace.getLibertyVersion()!= null && !libertyWorkspace.getLibertyVersion().isEmpty() &&
+        libertyWorkspace.getLibertyRuntime()!= null && !libertyWorkspace.getLibertyRuntime().isEmpty()) {
+      xsdDestFile = new File(tempDir, libertyWorkspace.getLibertyRuntime() + "-" + libertyWorkspace.getLibertyVersion() + ".xsd");
+    }
+
+    if (!xsdDestFile.exists()) {
+      try {
+        String xsdDestPath = xsdDestFile.getCanonicalPath();
+  
+        LOGGER.info("Generating schema file at: " + xsdDestPath);
+  
+        ProcessBuilder pb = new ProcessBuilder("java", "-jar", schemaGenJarPath.toAbsolutePath().toString(), xsdDestPath); //Add locale param here
+        pb.directory(tempDir);
+        pb.redirectErrorStream(true);
+        pb.redirectOutput(new File(tempDir, "schemagen.log"));
+  
+        Process proc = pb.start();
+        if (!proc.waitFor(30, TimeUnit.SECONDS)) {
+          proc.destroy();
+          throw new Exception("Exceeded 30 second timeout during schema file generation. Using cached schema.xsd file.");
```

Comment:
If this exception is thrown, the calling method does not default to using the cached schema.xsd. It probably should.

Reply:
Removed the exception, added a log message, and return null here now.

The catch block was returning null when this exception was thrown, but it wasn't very clear.